src/foam/u2/tag/Select.js

```diff
@@ -97,10 +104,12 @@ foam.CLASS({
         for ( var i = 0 ; i < choices.length ; i++ ) {
           var c = choices[i];
           let value = c[1];
+          var isDisabled = self.slot(function(data) { return self.isOptionDisabled_(value, data === i); });
           let e = self.E('option').attrs({
             value: i,
-            selected: self.data === i
-          }).translate(c[1] + '.name', value)
+            selected: self.data === i,
+            disabled: isDisabled
```

Comment:
Not a blocker, could make the intention to disable reveals itself instead of passing isSelected flag in the function.
```suggestion
            disabled: isSelected || isOptionDisabled_(value);
```
The `isOptionDisabled_` function can also be replaced with `self.disabled_data.some(o => foam.util.equals(o, value))`.

Reply:
Adapted your suggestion, but it works well with no slot for data